lib/feed.js

```diff
@@ -170,16 +170,43 @@ Feed.prototype.has = function (block) {
   return this.bitfield.get(block)
 }
 
-Feed.prototype.get = function (block, cb) { // TODO: on static feeds return null if > blocks
+Feed.prototype.get = function (block, opts, cb) { // TODO: on static feeds return null if > blocks
+  if (typeof opts === 'function') return this.get(block, null, opts)
+  if (!opts) opts = {}
   var self = this
 
   if (!this.opened) return this._openAndGet(block, cb)
 
-  if (!this.bitfield.get(block)) return this._getNext(block, cb)
+  if (!this.bitfield.get(block)) {
+    return this._getNext(block, cb)
+  }
 
-  if (this.storage) this.storage.get(block, done)
+  if (this.storage) this.storage.get(block, doneStorage)
   else this._core._data.get(this._prefix + block, done)
 
+  function doneStorage (err, value) {
+    if (err) return cb(err)
+    // there are different storage schemes which may not keep historical data
+    // if opts.verify is enabled, we need to double check that the data returned
+    // is correct. if it's not, then we must update the local bitfield and
+    // return an error.
+    if (opts.verify) {
+      self._core._nodes.get(self._prefix + (block * 2), function (err, node) {
+        if (err) return cb(err)
+        var blockHash = hash.data(value)
+        if (!equals(blockHash, node.hash)) {
+          self.bitfield.set(block, false) // remove from the local bitfield
+          self._sync()
+          cb(new Error('Block not found'))
+        } else {
+          done(null, value)
+        }
+      })
+    } else {
+      done(null, value)
+    }
```

Comment:
you can same a lot of indentation by doing `if (!opts.verify) return done(null, value)` first